builds/e2e/templates/lock-test-agents.yaml

```diff
@@ -7,12 +7,9 @@ jobs:
     displayName: Lock nested agents for ${{ parameters['upstream.protocol'] }}
     timeoutInMinutes: 180
     pool:
-     name: $(pool.name)
-     demands:
-       - agent-group -equals $(agent.group)
-       - Agent.OS -equals Linux
-       - Agent.OSArchitecture -equals X64
-       - status -equals unlocked
```

Comment:
I didn't follow the point you made in the description about not trying to obtain an unlocked agent. Can you explain some more?

Reply:
If we try to obtain an unlocked agent, sometimes all the agents will be locked. This is bad because devops never waits for an agent with certain capabilities to become available. It will immediately fail the build.

Switching to 1ES fixes this because there will always be agents available to wait on an up-front agent lock.